### DISCUSSION TITLE
Trino doesn't allow to create a role in hive catalog

### DISCUSSION THREAD

**Discussion by hsushmitha** [trinodb/trino#23276] [General]
Hi, 
   We are trying to create custom roles in Trino in hive catalog to restrict users from executing certain queries. 
We are executing this query through trino rest api: /v1/statement:
`CREATE ROLE customRole In hive`
we get this error:
`This connector does not support roles`
do we have an alternative to restrict users from executing certain queries eg: delete queries.
Ideally we want to create role and map users to the roles and then grant privileges on these roles, but it looks like hive connector doesn't support role creation.

Thanks,
Sushmitha

**Comment by ebyhr** [comment]
The connector supports creating roles. What `hive.security` are you using?

**Comment by hsushmitha** [comment]
we are setting `hive.security=allow-all` for hive catalog

**Comment by ebyhr** [comment]
The access control doesn't support creating roles. Why do you want to manage roles in allow-all security?

**Comment by Praveen2112** [comment]
Hive does support creating roles when we set the `hive.security=SQL_STANDARD` . Any specific reason not using that security mode in case of hive ?

**Comment by hsushmitha** [comment]
Initially we started off with not having roles. But now we have a scenario where we need to be able to restrict users to allow select, delete functionality.
so if we set `hive.security=SQL_STANDARD`, what roles would existing user assume?
